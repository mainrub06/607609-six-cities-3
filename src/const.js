export const STAR_PARAMS = {
  MIN_VALUE: 0,
  MAX_VALUE: 5,
  STAR_STEP: 20
};

export const LINKS = {
  INDEX: `/`,
  OFFER_DETAIL: `/offer/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`
};

export const OFFERS_SORT_ITEMS = [`Popular`, `Price: low to high`, `Price: high to low`, `Top rated first`];

export const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const MAX_PHOTOS_OFFER_DETAIL = 6;

export const AUTHORIZATION_STATUS = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

export const FAVORITE_REQUESTS = {
  ADD: `1`,
  DELETE: `0`
};

export const REQUEST_STATUS = {
  NO_AUTH: 401,
  OK: 200
};

export const OFFERS_CSS_CLASSES = {
  MAIN: {
    LIST: `cities__places-list`,
    ITEM: `cities__place-card`,
    IMAGE_WRAPPER: `cities__image-wrapper`,
    ITEM_INFO: ``,
    IMAGE_SIZE: {
      WIDTH: 260,
      HEIGHT: 200
    }
  },
  OFFER_DETAIL: {
    LIST: `near-places__list`,
    ITEM: `near-places__card`,
    IMAGE_WRAPPER: `near-places__image-wrapper`,
    ITEM_INFO: ``,
    IMAGE_SIZE: {
      WIDTH: 260,
      HEIGHT: 200
    }
  },
  FAVORITE: {
    LIST: `favorites__places`,
    ITEM: `favorites__card`,
    IMAGE_WRAPPER: `favorites__image-wrapper`,
    ITEM_INFO: `favorites__card-info`,
    IMAGE_SIZE: {
      WIDTH: 150,
      HEIGHT: 110
    }
  }
};

export const DETAIL_PAGE_PARAMS = {
  NEAR_OFFERS_MAX: 3
};

export const FORM_PARAMS = {
  STARS: [`perfect`, `good`, `not bad`, `badly`, `terribly`],
  MIN_TEXTAREA_VALUE: 50,
  MAX_TEXTAREA_VALUE: 300,
  MIN_STARS_CHECKED: 1
};

export const REVIEWS_PARAMS = {
  MAX_REVIEWS: 10
};
