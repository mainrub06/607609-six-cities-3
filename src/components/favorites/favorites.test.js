import React from "react";
import renderer from "react-test-renderer";
import Favorites from "./favorites.jsx";
import PLACE_DETAIL_MOCK from "../../mock/offers-detail";
import { OFFERS_CSS_CLASSES, CITIES } from "../../const";
import { MemoryRouter } from "react-router-dom";

const userInfo = {
  userAvatar: `img/1.png`,
  userEmail: `Oliver.conner@gmail.com`,
  id: 1,
  isPro: false,
  userName: `Oliver.conner`,
};

const favoriteResponse = true;

it(`Render snapshot <Favorite/>`, () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Favorites
          getFavoritesServerData={() => {}}
          citiesNames={CITIES}
          favorites={PLACE_DETAIL_MOCK}
          userInfo={userInfo}
          favoriteResponse={favoriteResponse}
          handleClickFavoriteButton={() => {}}
          onOfferClick={() => {}}
          offersCssClasses={OFFERS_CSS_CLASSES.FAVORITE}
        />
      </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
