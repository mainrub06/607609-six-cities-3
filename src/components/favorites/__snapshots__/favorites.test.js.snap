// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Favorite component 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="main.html"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Dusseldorf
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places places__list"
            />
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
